protocol/x/affiliates/keeper/keeper_test.go

```diff
@@ -21,141 +23,174 @@ func TestLogger(t *testing.T) {
 	require.NotNil(t, logger)
 }
 
-func TestRegisterAffiliate(t *testing.T) {
-	tApp := testapp.NewTestAppBuilder(t).Build()
-	ctx := tApp.InitChain()
-	k := tApp.App.AffiliatesKeeper
-
+func TestRegisterAffiliate_GetReferredBy(t *testing.T) {
 	tests := []struct {
 		name        string
 		referee     string
 		affiliate   string
 		expectError bool
+		setup       func(t *testing.T, ctx sdk.Context, k *keeper.Keeper)
 	}{
 		{
 			name:        "Register new affiliate",
-			referee:     "referee1",
-			affiliate:   "affiliate1",
+			referee:     constants.AliceAccAddress.String(),
+			affiliate:   constants.BobAccAddress.String(),
 			expectError: false,
+			setup: func(t *testing.T, ctx sdk.Context, k *keeper.Keeper) {
+				// No setup needed for this test case
+			},
 		},
 		{
 			name:        "Register existing referee",
-			referee:     "referee1",
-			affiliate:   "affiliate2",
+			referee:     constants.AliceAccAddress.String(),
+			affiliate:   constants.CarlAccAddress.String(),
 			expectError: true,
+			setup: func(t *testing.T, ctx sdk.Context, k *keeper.Keeper) {
+				err := k.RegisterAffiliate(ctx, constants.AliceAccAddress.String(), constants.BobAccAddress.String())
+				require.NoError(t, err)
+			},
 		},
 	}
 
 	for _, tc := range tests {
+		tApp := testapp.NewTestAppBuilder(t).Build()
+		ctx := tApp.InitChain()
+		k := tApp.App.AffiliatesKeeper
+		tc.setup(t, ctx, &k)
 		t.Run(tc.name, func(t *testing.T) {
 			err := k.RegisterAffiliate(ctx, tc.referee, tc.affiliate)
 			if tc.expectError {
 				require.Error(t, err)
```

Comment:
Nit: Use `require.ErrorIs()` or `require.ErrorContains()` to test error more granularly, instead of a binary result of whether it errors or not.